Add host authorization middleware options to the config guide [ci-skip] (#42825)
rails/rails: guides/source/configuring.md

    Rails.application.config.hosts << ".product.com"
    ```

>     You can exclude certain requests from Host Authorization checks with `exclude`:

Reviewer: ```suggestion
    You can exclude certain requests from Host Authorization checks by setting 
    `config.host_configuration.exclude`:
```

    ```ruby
    # Exclude requests for the /healthcheck/ path from host checking

>     Rails.application.config.host_configuration = { exclude: ->(request) { request.path =~ /healthcheck/ } }

Reviewer: ```suggestion
    Rails.application.config.host_configuration = { 
      exclude: ->(request) { request.path =~ /healthcheck/ } 
    }
```

    ``` 

>     When a request comes to an unauthorized host, a default response application will run and respond with a `403 Forbidden`. This response can be customized using `config.host_configuration.response_app`. For an example of a response application, see the [ActionDispatch::HostAuthorization API documentation](https://api.rubyonrails.org/classes/ActionDispatch/HostAuthorization.html).

Reviewer: Since the text wrapping of `DEFAULT_RESPONSE_APP` at https://api.rubyonrails.org/classes/ActionDispatch/HostAuthorization.html makes it difficult to read, what if we show an extremely basic example here instead:

```suggestion
    When a request comes to an unauthorized host, a default Rack application 
    will run and respond with `403 Forbidden`. This can be customized by setting 
    `config.host_configuration.response_app`. For example:

    ```ruby
    Rails.application.config.host_configuration = {
      response_app: -> env do
        [400, { "Content-Type" => "text/plain" }, ["Bad Request"]]
      end
    }
    ```
```
